### PR TITLE
Implement trusted advisor addons recommendations

### PR DESCRIPTION
**Pull request by gabelton**
https://uktrade.atlassian.net/browse/DBTP-308

Fixing various trusted advisor recommendations and a few bugs encountered when redeploying return-to-office.

Recommendations and action taken:
- Enable bucket logging for most buckets - adding a logging bucket and policy to s3 template
- Enable bucket versioning on most buckets to prevent accidental deletion or overwriting - updated s3 template
- S3 Block Public Access setting should be enabled at the bucket-level - set `IgnorePublicAcls` and `RestrictPublicBuckets` to true in s3 template
- Amazon ECS Containers should only have read-only access to its root filesystems - set `storage.readonly_fs: true` in public and backend templates

Bugs with rds template:
- `MaxAllocatedStorage` needs to be an int
- `EngineVersion` 13.4 not supported

### CODE REVIEW

**Comment by WillGibson** [comment]
Looks like you have conflicts to resolve @gabelton.

Also, how has this been tested?

File: templates/addons/env/rds-postgres.yml [hunk 1]
```diff
@@ -73,7 +73,7 @@ Resources:
       AutoMinorVersionUpgrade: true
       EnablePerformanceInsights: true
       Engine: postgres
-      EngineVersion: '13.4'
+      EngineVersion: '13'
```
Comment (WillGibson) [inline templates/addons/env/rds-postgres.yml:76]: Not for this pull request, but do we want to make this something that developer teams can specify themselves?

File: templates/addons/env/s3.yml [hunk 2]
```diff
@@ -19,6 +16,51 @@ Mappings:
     {% endfor %}
 
 Resources:
+  {{ service.prefix }}LoggingBucket:
+    Metadata:
+      'aws:copilot:description': 'An Amazon S3 bucket to store logs from {{ service.prefix }}Bucket'
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Join [-, [!FindInMap [{{ service.prefix }}BucketNameMap, !Ref Env, BucketName], logs]]
```
Comment (WillGibson) [inline templates/addons/env/s3.yml:24]: `!Join [-, [!FindInMap [{{ service.prefix }}BucketNameMap, !Ref Env, BucketName], logs]]` appears twice in this template. I wonder if we can set it in a parameter to remove the risk that we might change it in one place, but not the other?

**Comment by yusufsheiqh** [comment]
To resolve the conflicts... you will have to rebase `main` and make sure the `templates` folder is inside the `commands` folder.

File: templates/addons/env/rds-postgres.yml (hunk 1 above)
Comment (WillGibson) [reply templates/addons/env/rds-postgres.yml:76]: [DBTP-336 Allow overriding of default Postgres version](https://uktrade.atlassian.net/browse/DBTP-336)

**Comment by gabelton** [comment]
I've resolved merge conflicts and, in terms of testing, I had to fix one of the `make_config` tests and tested manually by redeploying return to office and adding s3 buckets. I think there's probably a gap in our unit testing, though, because more tests should fail when I make several changes to rds and s3 templates. I don't know whether that belongs in this ticket or not though

**Comment by gabelton** [comment]
I've created a ticket for adding more unit tests here https://uktrade.atlassian.net/browse/DBTP-338

**Review by WillGibson — APPROVED** [review]

**Pull request merged**